src/platform/plugins/shared/controls/public/control_group/initialize_editor_state_manager.ts

```diff
@@ -7,20 +7,25 @@
  * License v3.0 only", or the "Server Side Public License, v 1".
  */
 
+import type {
+  ControlsChainingSystem,
+  ControlsIgnoreParentSettings,
+  ControlsLabelPosition,
+} from '@kbn/controls-schemas';
 import { StateComparators, initializeStateManager } from '@kbn/presentation-publishing';
-import { ControlGroupEditorState } from './types';
 import {
   DEFAULT_AUTO_APPLY_SELECTIONS,
-  DEFAULT_CONTROL_CHAINING,
-  DEFAULT_CONTROL_LABEL_POSITION,
+  DEFAULT_CONTROLS_CHAINING,
+  DEFAULT_CONTROLS_LABEL_POSITION,
   DEFAULT_IGNORE_PARENT_SETTINGS,
-} from '../../common';
+} from '@kbn/controls-constants';
+import { ControlGroupEditorState } from './types';
 
 export const defaultEditorState = {
   autoApplySelections: DEFAULT_AUTO_APPLY_SELECTIONS,
-  chainingSystem: DEFAULT_CONTROL_CHAINING,
-  ignoreParentSettings: DEFAULT_IGNORE_PARENT_SETTINGS,
-  labelPosition: DEFAULT_CONTROL_LABEL_POSITION,
+  chainingSystem: DEFAULT_CONTROLS_CHAINING as ControlsChainingSystem,
```

Comment:
Is it possible at all for the `controls-constants` package to import these types from the `controls-schemas` package? This way we don't have to cast them here.

Reply:
At the moment, no. That would cause a circular reference issue as `controls-schemas` has a dependency on `controls-constants` for setting defaults. But in a future PR we could probably just completely get rid of these default constants. We would not set any defaults in the schemas and the UI would be the only place we define default values. Then, if the value is undefined in the saved object, the UI would determine the default.